plugins/newsdownloader.koplugin/main.lua

```diff
@@ -163,89 +247,188 @@ function NewsDownloader:loadConfigAndProcessFeeds()
     UI:info("Loading news feed config…")
     logger.dbg("force repaint due to upcoming blocking calls")
 
-    local ok, feed_config = pcall(dofile, feed_config_path)
+    local ok, feed_config = pcall(dofile, self.feed_config_path)
     if not ok or not feed_config then
         UI:info(T(_("Invalid configuration file. Detailed error message:\n%1"), feed_config))
         return
     end
-
+    -- If the file contains no table elements, then the user hasn't set any feeds
     if #feed_config <= 0 then
-        logger.err('NewsDownloader: empty feed list.', feed_config_path)
+        logger.err('NewsDownloader: empty feed list.', self.feed_config_path)
+        -- TODO: Ask them to set the first feed. Perhaps even suggest something?!
         return
     end
 
-    local never_download_images = news_downloader_settings:isTrue("never_download_images")
-
+    local never_download_images = self.settings:isTrue("never_download_images")
     local unsupported_feeds_urls = {}
-
     local total_feed_entries = #feed_config
+    local feed_message
+
     for idx, feed in ipairs(feed_config) do
         local url = feed[1]
         local limit = feed.limit
         local download_full_article = feed.download_full_article == nil or feed.download_full_article
         local include_images = not never_download_images and feed.include_images
         local enable_filter = feed.enable_filter or feed.enable_filter == nil
         local filter_element = feed.filter_element or feed.filter_element == nil
+        -- Check if the two required attributes are set
         if url and limit then
-            local feed_message = T(_("Processing %1/%2:\n%3"), idx, total_feed_entries, BD.url(url))
+            feed_message = T(_("Processing %1/%2:\n%3"), idx, total_feed_entries, BD.url(url))
             UI:info(feed_message)
-            NewsDownloader:processFeedSource(url, tonumber(limit), unsupported_feeds_urls, download_full_article, include_images, feed_message, enable_filter, filter_element)
+            -- Process, i.e.: "Download"
+            self:processFeedSource(
+                url,
+                tonumber(limit),
+                unsupported_feeds_urls,
+                download_full_article,
+                include_images,
+                feed_message,
+                enable_filter,
+                filter_element)
         else
-            logger.warn('NewsDownloader: invalid feed config entry', feed)
+            logger.warn('NewsDownloader: invalid feed config entry.', feed)
```

Comment:
If you change the line for minor nitpicks, nitpick in full. ;-)
```suggestion
            logger.warn("NewsDownloader: invalid feed config entry.", feed)
```